### PR TITLE
Add support for pyOCD

### PR DESCRIPTION
**Pull request by c1728p9**
Add copy and reset support using pyOCD.

### CODE REVIEW

**Comment by bridadan** [comment]
@c1728p9 Could you please add pyocd to the requirements?

https://github.com/ARMmbed/htrun/blob/master/setup.py#L55

That may help with the CircleCI error as well.

**Comment by c1728p9** [comment]
This PR requires an updated version of pyOCD so don't merge it yet.

File: mbed_host_tests/host_tests_plugins/module_copy_pyocd.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                rom_region = memory_map.getBootMemory()
+                with open(image_path, "rb") as file_handle:
+                    program_data = file_handle.read()
+                program_list.append((rom_region.start, program_data))
+            elif extension == '.hex':
+                # Intel hex file format
+                ihex = IntelHex(image_path)
+                addresses = ihex.addresses()
+                addresses.sort()
+                data_list = list(_ranges(addresses))
```
Comment (mazimkhan) [inline mbed_host_tests/host_tests_plugins/module_copy_pyocd.py]: Perhaps some comment here saying "Get contiguous address ranges with start and end address".

Comment (mazimkhan) [inline mbed_host_tests/host_tests_plugins/module_copy_pyocd.py]: Also why convert it into a list when a generator is returned by ```_ranges```.

**Comment by mazimkhan** [comment]
LGTM apart from minor documentation requirements.

**Comment by c1728p9** [comment]
Before this PR can be merged pyOCD must support simultaneous device access:
https://github.com/mbedmicro/pyOCD/pull/268

**Comment by c1728p9** [comment]
Updated and rebased PR. The changes to pyOCD have been merged so this is good to go from my end.

**Comment by adbridge** [comment]
@mazimkhan please re-review

**Comment by mazimkhan** [comment]
LGTM

**Pull request merged**